library/leon/io/StdIn.scala

```diff
@@ -16,38 +16,38 @@ object StdIn {
   @isabelle.noBody()
   def readInt(implicit state: State): Int = {
     state.seed += 1
-    nativeReadInt
+    nativeReadInt(state.seed)
   }
 
   @library
   @extern
-  private def nativeReadInt(implicit state: State): Int = {
+  private def nativeReadInt(seed: BigInt): Int = {
```

Comment:
Hey. Actually we found out that if we were to pass state to an @extern function, the verification would have to assume it could be mutated (because extern disregards the body, and we should be sound). That's unfortunate as we would like to make sure each call to nativeXXX are done with distinct arguments. Passing the BigInt explicitly from the State in the non-native function ensures that the value is different each time.